### PR TITLE
61 add paggination for user dirictory

### LINKED ISSUE #61: Add paggination for user dirictory

### PR DESCRIPTION
**Pull request by BtsuanEyakem**

### CODE REVIEW

File: src/main/java/edu/miu/cs545/project/controller/UserController.java [hunk 1]
```diff
@@ -3,15 +3,28 @@
...
 
+    private final UserService userService;
     public UserController(UserService userService) {
         super(userService);
+        this.userService = userService;
+    }
+
+    public ResponseEntity<Page<User>> getUserByPage(@RequestParam int pageSize){
+        Page<User> userPage = userService.getUsersByPage(pageSize);
+        return null;
```
Comment (darkhanzhursin) [inline src/main/java/edu/miu/cs545/project/controller/UserController.java]: avoid returning null

File: src/main/java/edu/miu/cs545/project/service/impl/SurveyViewServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.stereotype.Service;
+
+@Service
+public class SurveyViewServiceImpl extends CrudServiceImpl<SurveyView,Long> implements SurveyViewServce {
+    public SurveyViewServiceImpl(SurveyViewRepo repository) {
+        super(repository);
+    }
+}
+
+
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/service/impl/SurveyViewServiceImpl.java:18]: let's remove extra empty lines

File: src/main/java/edu/miu/cs545/project/mapper/impl/StudentDtoMapper.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package edu.miu.cs545.project.mapper.impl;
+
+import edu.miu.cs545.project.dto.StudentDTO;
+import edu.miu.cs545.project.model.entity.Student;
+
+import java.time.LocalDate;
+
+public class StudentDtoMapper {
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/mapper/impl/StudentDtoMapper.java:8]: Use MapStruct

File: src/main/java/edu/miu/cs545/project/service/UserService.java [hunk 4]
```diff
@@ -1,6 +1,8 @@
 package edu.miu.cs545.project.service;
 
 import edu.miu.cs545.project.model.entity.User;
+import org.springframework.data.domain.Page;
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/service/UserService.java:4]: remove unused import

File: src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java [hunk 5]
```diff
@@ -3,11 +3,64 @@
...
+        }
+
+        if (major != null && !major.isEmpty()) {
+            predicates.add(cb.equal(cb.lower(studentDirectory.get("major")), major.toLowerCase()));
+        }
+
+        if (otherFilter != null && !otherFilter.isEmpty()) {
+            predicates.add(cb.like(cb.lower(studentDirectory.get("contactInformation")), "%" + otherFilter.toLowerCase() + "%"));
+        }
+
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java:54]: if no data coming form contactInformation we need to search in student object, first name, last name at least

File: src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java [hunk 6]
```diff
@@ -3,11 +3,64 @@
...
+        cq.where(predicates.toArray(new Predicate[0]));
```
Comment (nabil-miu) [inline src/main/java/edu/miu/cs545/project/service/impl/StudentDirectoryServiceImpl.java:55]: test

**Review by reza-newgen — COMMENTED** [review]
@GetMapping("/pages/{page}")
    public ResponseEntity<List<StudentDirectoryDTO>> getPage(@PathVariable int page, @RequestParam(required = false) Integer size) {
        if (size == null) {
            size = 10;
        }   

endpoints are join each other. please make one line space above the endpoint.

**Pull request closed without merging**